client/src/components/_common/TextCopyField/TextCopyField.jsx

```diff
@@ -30,44 +30,74 @@ const TextCopyField = ({ value, placeholder }) => {
 
   return (
     <div className="input-group">
-      <div className="input-group-prepend">
-        <CopyToClipboard text={value}>
-          <Button
-            className={styles['copy-button']}
-            // RFE: Avoid manual JS ↔ CSS sync of transition duration by using:
-            //      - `data-attribute` and `attr()` (pending browser support)
-            //      - PostCSS and JSON variables (pending greater need for this)
-            onClick={onCopy}
-            disabled={isEmpty}
-            type="secondary"
-            size="medium"
-            iconNameBefore={isCopied ? 'approved-reverse' : 'link'}
-          >
-            Copy
-          </Button>
-        </CopyToClipboard>
-      </div>
-      <input
-        type="text"
-        onChange={onChange}
-        value={value}
-        className={`form-control ${styles.input}`}
-        placeholder={placeholder}
-        data-testid="input"
-        readOnly
-      />
+      {renderType === 'textarea' ? (
+        <>
+          <textarea
+            onChange={onChange}
+            value={value}
+            className={`${styles.textarea}`}
+            placeholder={placeholder}
+            data-testid="textarea"
+            readOnly
+          />
+          <div className="text-right mt-3">
+            <CopyToClipboard text={value}>
+              <Button
+                onClick={onCopy}
+                disabled={isEmpty}
+                type="secondary"
+                size="medium"
+                iconNameBefore={isCopied ? 'approved-reverse' : 'link'}
+              >
+                Copy
+              </Button>
+            </CopyToClipboard>
+          </div>
+        </>
+      ) : (
+        <>
+          <div className="input-group-prepend">
+            <CopyToClipboard text={value}>
+              <Button
+                className={styles['copy-button']}
+                // RFE: Avoid manual JS ↔ CSS sync of transition duration by using:
+                //      - `data-attribute` and `attr()` (pending browser support)
+                //      - PostCSS and JSON variables (pending greater need for this)
+                onClick={onCopy}
+                disabled={isEmpty}
+                type="secondary"
+                size="medium"
+                iconNameBefore={isCopied ? 'approved-reverse' : 'link'}
+              >
+                Copy
+              </Button>
+            </CopyToClipboard>
+          </div>
+          <input
+            type="text"
+            onChange={onChange}
+            value={value}
+            className={`form-control ${styles.input}`}
+            placeholder={placeholder}
+            data-testid="input"
+            readOnly
+          />
+        </>
+      )}
```

Comment:
I like the solution, but can this be done with less duplicate code.

All I can think of is a mini internal component for the `<CopyToClipboard />`. And, if the usage was identical, such a mini-component would not need a prop.

Reply:
Made an object to hold common prop types between the copy button and the two fields. A lot less code/space used now. Retains same functionality.

Reply:
It looks good. Thanks. 👍